src/libp2p_config.erl

```diff
@@ -247,6 +249,35 @@ lookup_session_addrs(TID, Pid) ->
 -spec lookup_session_addrs(ets:tab()) -> [string()].
 lookup_session_addrs(TID) ->
     lookup_addrs(TID, ?SESSION).
+
+-spec lookup_session_addr_info(ets:tab(), pid()) -> {ok, {string(), string()}} | false.
+lookup_session_addr_info(TID, Pid) ->
+    lookup_addr_info(TID, ?SESSION, Pid).
+
+-spec insert_session_addr_info(ets:tab(), pid(), {string(), string()}) -> true.
+insert_session_addr_info(TID, Pid, AddrInfo) ->
+    insert_addr_info(TID, ?SESSION, Pid, AddrInfo).
+
+
+%%
+%% Addr Info
+%%
+
+-spec insert_addr_info(ets:tab(), atom(), pid(), {string(), string()}) -> true.
+insert_addr_info(TID, Kind, Pid, AddrInfo) ->
+    %% Insert in the form that remove_pid understands to ensure that
+    %% addr info gets removed for removed pids regardless of what kind
+    %% of addr_info it is
+    ets:insert(TID, {{?ADDR_INFO, Kind, AddrInfo}, Pid}).
+
+-spec lookup_addr_info(ets:tab(), atom(), pid()) -> {ok, {string(), string()}} | false.
+lookup_addr_info(TID, Kind, Pid) ->
+    case ets:lookup(TID, {{?ADDR_INFO, Kind, '$1'}, Pid}) of
```

Comment:
Should we do an is_process_alive check too?

Reply:
We don't do it anywhere else either?

remove_pid causes the addr_info entries to be removed from the table, and that's called by the swarm_server when the session goes away (monitored)